tests/terminator_tests/improvement_tests/test_evaluator.py

```diff
@@ -1,45 +1,99 @@
-from typing import List
-from typing import Tuple
+from __future__ import annotations
+
 from unittest import mock
 
-import numpy as np
 import pytest
 
 from optuna.distributions import FloatDistribution
 from optuna.study import StudyDirection
 from optuna.terminator.improvement._preprocessing import NullPreprocessing
 from optuna.terminator.improvement.evaluator import RegretBoundEvaluator
-from optuna.terminator.improvement.gp.base import _get_beta
-from optuna.terminator.improvement.gp.base import BaseGaussianProcess
 from optuna.trial import create_trial
 from optuna.trial import FrozenTrial
 
 
-class _StaticGaussianProcess(BaseGaussianProcess):
-    """A dummy BaseGaussianProcess class always returning 0.0 mean and 1.0 std
-
-    This class is introduced to make the GP class and the evaluator class loosely-coupled in unit
-    testing.
-    """
-
-    def fit(
-        self,
-        trials: List[FrozenTrial],
-    ) -> None:
-        pass
-
-    def predict_mean_std(self, trials: List[FrozenTrial]) -> Tuple[np.ndarray, np.ndarray]:
-        mean = np.zeros(len(trials))
-        std = np.ones(len(trials))
-
-        return mean, std
-
-
 # TODO(g-votte): test the following edge cases
 # TODO(g-votte): - the user specifies non-default top_trials_ratio or min_n_trials
 
 
-def test_evaluate() -> None:
+@pytest.mark.parametrize(
```

Comment:
How about adding test cases to validate the `IntDistribution`, `Float/IntDistribution(log=True)` and `CategoricalDistribution`. This might be worked on the follow-up PRs.

Reply:
Done.